src/main/java/org/example/bookshop/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,134 @@
+            orderItemRepository.save(item);
+        });
+        order.setOrderItems(orderItems);
+        shoppingCart.getCartItems().clear();
+        cartRepository.save(shoppingCart);
+        return orderMapper.toDto(order);
+    }
+
+    @Override
+    public Set<OrderDto> getAll(User user, Sort sort, Pageable pageable) {
+        return orderRepository.findAllByUserId(user.getId(), pageable).stream()
+                .map(orderMapper::toDto)
+                .collect(Collectors.toSet());
+    }
+
+    @Override
+    public void update(OrderStatusRequestDto requestDto, Long orderId, User user) {
+        Order order = findOrderByIdAndUserId(orderId, user.getId());
+        Order.Status status = Order.Status.valueOf(requestDto.getStatus());
+        if (order.getStatus().equals(status)) {
+            throw new DataProcessingException("Order already has status: " + status);
+        }
+        order.setStatus(status);
+        orderRepository.save(order);
+    }
+
+    @Override
+    public Set<OrderItemDto> get(Long orderId, User user) {
+        findOrderByIdAndUserId(orderId, user.getId());
+        return findOrderItems(user).stream()
+                .map(orderItemMapper::toDto)
+                .collect(Collectors.toSet());
+        // Have alternative solve, it should be faster,
+        // but need 1 more method in orderItemRepository
+        // return orderItemRepository.findAllByOrderId(orderId)
+//                .stream()
+//                .filter(Optional::isPresent)
+//                .map(Optional::get)
+//                .map(orderItemMapper::toDto)
+//                .collect(Collectors.toSet());
+    }
+
+    @Override
+    public OrderItemDto getOrderItem(Long orderId, Long itemId, User user) {
+        findOrderByIdAndUserId(orderId, user.getId());
+        Optional<OrderItem> itemOptional = orderItemRepository.findByIdAndOrderId(itemId, orderId);
+        if (itemOptional.isEmpty()) {
+            throw new EntityNotFoundException("Can't find item with id: " + itemId);
+        }
+        return orderItemMapper.toDto(itemOptional.get());
+    }
+
+    private Order findOrderByIdAndUserId(Long orderId, Long userId) {
+        Optional<Order> optionalOrder = orderRepository.findByIdAndUserId(orderId, userId);
+        if (optionalOrder.isEmpty()) {
+            throw new EntityNotFoundException("Can't find order with id: " + orderId);
+        }
+        return optionalOrder.get();
+    }
+
+    private Set<OrderItem> findOrderItems(User user) {
+        Set<OrderItem> orderItem = cartItemRepository.findAllByShoppingCartId(
+                cartRepository.findByUserId(user.getId()).get().getId())
+                .stream()
+                .map(orderItemMapper::toEntity)
+                .collect(Collectors.toSet());
+        if (orderItem.isEmpty()) {
+            throw new EntityNotFoundException(
+                    "Can't find order items for user: " + user.getEmail());
+        }
```

Comment:
why should it occur Exception?

Reply:
This check doesn't allow to place an empty order. Added more description into exception